### PR TITLE
Update doc

### PR DESCRIPTION
**Pull request by simonpasquier**
This PR fixes a couple of typos in the format documentation. It also references this documentation from the main README file.

### CODE REVIEW

File: Documentation/format/index.md [hunk 1]
```diff
@@ -139,11 +139,11 @@ The `#names` field determines the number indexed label names, followed by the to
 └─────────────────────────────────────────────────┘
 ```
 
-The sequence of label index sections is finalized by an offset table pointing to the beginning of each label index section for a given set of label names.
+The sequence of label index sections is finalized by an [offset table](#offset-table) pointing to the beginning of each label index section for a given set of label names.
```
Comment (shubheksha) [inline Documentation/format/index.md]: The wording needs to be fixed here. "is finalized by" makes it sound like the section itself ends with an offset table which isn't the case.

File: Documentation/format/index.md [hunk 2]
```diff
@@ -161,7 +161,7 @@ Postings sections store monotonically increasing lists of series references that
 └─────────────────────────────────────────┘
 ```
 
-The sequence of postings sections is finalized by an offset table pointing to the beginning of each postings section for a given set of label names.
+The sequence of postings sections is finalized by an [offset table](#offset-table) pointing to the beginning of each postings section for a given set of label names.
```
Comment (shubheksha) [inline Documentation/format/index.md]: Same as above for this too, the sentence needs rephrasing.

File: Documentation/format/index.md [hunk 3]
```diff
@@ -119,8 +119,8 @@ mint of the first chunk is stored and the mint and maxt are encoded as deltas to
 
 ### Label Index
 
-A label index section indexes the existing (combined) values for one or more label names.  
-The `#names` field determines the number indexed label names, followed by the total number of entries in the `#entries` field. The body holds `#entries` symbol table reference tuples of length of length `#names`. The value tuples are sorted in lexicographically increasing order.
+A label index section indexes the existing (combined) values for one or more label names.
+The `#names` field determines the number of indexed label names, followed by the total number of entries in the `#entries` field. The body holds `#entries` symbol table reference tuples of length `#names`. The value tuples are sorted in lexicographically increasing order.
```
[the remark this answers is not in the document]
Comment (simonpasquier) [reply Documentation/format/index.md]: What about?

> the body holds `#entries / #names` tuples of symbol table references, each tuple being of `#names` length.

There could also be an concrete example.

For instance, a single label name with 4 different values will be encoded as:
```                                                                               
┌─────┬───┬───┬──────────────┬──────────────┬──────────────┬──────────────┬───────┐                                
│ len │ 1 │ 4 │ ref(value_0) | ref(value_1) | ref(value_2) | ref(value_3) | CRC32 |
└─────┴───┴───┴──────────────┴──────────────┴──────────────┴──────────────┴───────┘
```

**Comment by gouthamve** [comment]
LGTM! Thanks!

**Pull request merged**